Can not build projects on MacOS Mojave

**Describe the bug**
When trying to build a project using these rules, Bazel will attempt to download the necessary archives and packages to end up failing saying that it can't find the `@io_tweag_rules_haskell_ghc_darwin_amd64//` package.

Here is the standard out:

```bash
sed: 1: "mk/config.mk.in": invalid command code m
INFO: An error occurred during the fetch of repository 'io_tweag_rules_haskell_ghc_darwin_amd64'
INFO: Call stack for the definition of repository 'io_tweag_rules_haskell_ghc_darwin_amd64':
 - /private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl:364:5
 - /private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl:400:9
 - /Users/domingueza/Workspace/Projects/herd/WORKSPACE:35:1
INFO: Repository 'io_tweag_rules_haskell_ghc_darwin_amd64' used the following cache hits instead of downloading the corresponding file.
 * Hash 'cccb58f142fe41b601d73690809f6089f7715b6a50a09aa3d0104176ab4db09e' for https://downloads.haskell.org/~ghc/8.6.4/ghc-8.6.4-x86_64-apple-darwin.tar.xz
If the definition of 'io_tweag_rules_haskell_ghc_darwin_amd64' was updated, verify that the hashes were also updated.
ERROR: /private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell_ghc_darwin_amd64-toolchain/BUILD:4:1: no such package '@io_tweag_rules_haskell_ghc_darwin_amd64//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl", line 215
		_execute_fail_loudly(ctx, ["sed", "-i", "s/RelocatableB..."])
	File "/private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl", line 177, in _execute_fail_loudly
		fail("{0} failed, aborting creation o...)))
sed -i s/RelocatableBuild = NO/RelocatableBuild = YES/ mk/config.mk.in failed, aborting creation of GHC bindist and referenced by '@io_tweag_rules_haskell_ghc_darwin_amd64-toolchain//:toolchain-impl'
ERROR: Analysis of target '//protocol:protocol' failed; build aborted: no such package '@io_tweag_rules_haskell_ghc_darwin_amd64//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl", line 215
		_execute_fail_loudly(ctx, ["sed", "-i", "s/RelocatableB..."])
	File "/private/var/tmp/_bazel_domingueza/cf40c2340e1460b987f62a06aab24e21/external/io_tweag_rules_haskell/haskell/ghc_bindist.bzl", line 177, in _execute_fail_loudly
		fail("{0} failed, aborting creation o...)))
sed -i s/RelocatableBuild = NO/RelocatableBuild = YES/ mk/config.mk.in failed, aborting creation of GHC bindist
INFO: Elapsed time: 35.882s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (21 packages loaded, 238 targets configured)
```

**To Reproduce**
Clone the repo at `https://github.com/alonsodomin/herd` using the `bazel` branch and then run the Bazel build from the root folder:

```bash
git clone https://github.com/alonsodomin/herd
cd herd
git checkout bazel
bazel build --incompatible_bzl_disallow_load_after_statement=false //protocol
```

**Expected behavior**
Bazel should finish the build successfully.

**Environment**
 - OS name + version: macOS Mojave - 10.14.5
 - Bazel version: 0.25.2-homebrew
 - Version of the rules: 0.9

**Additional context**
You may also notice that with given version of Bazel, I need to use the `--incompatible_bzl_disallow_load_after_statement=false` parameter to accept the current rules as they are defined.

Did you do a `bazel clean --expunge`?

Is the error message the same? I’m getting a different error after applying the change:

```
protocol/src/Herd/Data/Text.hs:11:1: error:                                                                                                                                                                                                                                                 
    Could not load module _Data.Text_             
    It is a member of the hidden package _text-1.2.3.1_.        
    You can run _:set -package text_ to expose it.                                                                                                                                                                                                                                          
    (Note: this unloads all the modules in the current scope.)                                                                                                                                                                                                                              
    Use -v to see a list of the files searched for.                                                                                                                                                                                                                                     
   |                                           
11 | import           Data.Text        (Text)                                                                                                                                                                                                                                               
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^        
```

Which is to be expected, because the `text` package is not imported in the `BUILD.bazel` file (similar to how you have to specify them in `.cabal` files usually).

right, yeah, that looks like a legit error from the build config.

I was getting the same as before, even though I had run the clean cmd you mentioned (tried again today, same again)

Anyway, I may have something wrong in my setup so I’ll try to fix it. From what you showed here, it seems the issue is solved.

On vie, may 24, 2019 a las 18:51, Profpatsch <notifications@github.com> escribió:

> Did you do a bazel clean --expunge?
>
> Is the error message the same? I’m getting a different error after applying the change:
>
> protocol/src/Herd/Data/Text.hs:11:1: error:
>     Could not load module _Data.Text_
>     It is a member of the hidden package _text-1.2.3.1_.
>     You can run _:set -package text_ to expose it.
>     (Note: this unloads all the modules in the current scope.)
>     Use -v to see a list of the files searched for.
>    |
> 11 | import           Data.Text        (Text)
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> Which is to be expected, because the text package is not imported in the BUILD.bazel file (similar to how you have to specify them in .cabal files usually).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, [view it on GitHub](https://github.com/tweag/rules_haskell/issues/884?email_source=notifications&email_token=AARKCAFGR52T2LDHCPMJDRLPXAMINA5CNFSM4HOTRPZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODWF6JUA#issuecomment-495707344), or [mute the thread](https://github.com/notifications/unsubscribe-auth/AARKCAGBYN5HCRMMLB2FDO3PXAMINANCNFSM4HOTRPZA).

--- a/haskell/ghc_bindist.bzl
+++ b/haskell/ghc_bindist.bzl
@@ -215,15 +215,14 @@ def _ghc_bindist_impl(ctx):
 
     # On Windows the bindist already contains the built executables
     if os != "windows":
-        # IMPORTANT: all these scripts have to be compatible with BSD tools!
-
-        # bsdcompatible is to work around the differences for -i on BSD sed
-        _execute_fail_loudly(ctx, ["sed", "-e", "s/RelocatableBuild = NO/RelocatableBuild = YES/", "-ibsdcompatible", "mk/config.mk.in"])
+        # IMPORTANT: all these scripts have to be compatible with BSD
+        # tools! This means that sed -i always takes an argument.
+        _execute_fail_loudly(ctx, ["sed", "-e", "s/RelocatableBuild = NO/RelocatableBuild = YES/", "-i.bak", "mk/config.mk.in"])
         _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
         _execute_fail_loudly(ctx, ["make", "install"])
         ctx.file("patch_bins", executable = True, content = r"""#!/usr/bin/env bash
 grep --files-with-matches --null {bindist_dir} bin/* | xargs -0 -n1 \
-    sed -ibsdcompatible \
+    sed -i.bak \
         -e '2i\
           DISTDIR="$( dirname "$(resolved="$0"; while tmp="$(readlink "$resolved")"; do resolved="$tmp"; done; echo "$resolved")" )/.."' \
         -e 's:{bindist_dir}:$DISTDIR:'

--- a/haskell/ghc_bindist.bzl
+++ b/haskell/ghc_bindist.bzl
@@ -222,7 +222,7 @@ def _ghc_bindist_impl(ctx):
         _execute_fail_loudly(ctx, ["./configure", "--prefix", bindist_dir.realpath])
         _execute_fail_loudly(ctx, ["make", "install"])
         ctx.file("patch_bins", executable = True, content = r"""#!/usr/bin/env bash
-grep --files-with-matches --null {bindist_dir} bin/* | xargs -0 \
+grep --files-with-matches --null {bindist_dir} bin/* | xargs -0 -n1 \
     sed -ibsdcompatible \
         -e '2i\
           DISTDIR="$( dirname "$(resolved="$0"; while tmp="$(readlink "$resolved")"; do resolved="$tmp"; done; echo "$resolved")" )/.."' \

